src/uct/sm/mm/mm_iface.c

```diff
@@ -152,15 +153,48 @@ static inline void uct_mm_progress_fifo_tail(uct_mm_iface_t *iface)
     iface->recv_fifo_ctl->tail = iface->read_index;
 }
 
+ucs_status_t uct_mm_assign_desc_to_fifo_elem(uct_mm_iface_t *iface,
+                                     uct_mm_fifo_element_t *fifo_elem_p)
+{
+    uct_mm_recv_desc_t *desc;
+
+    UCT_TL_IFACE_GET_RX_DESC(&iface->super, iface->recv_desc_mp, desc,
```

Comment:
maybe this function should not return error code if there is no desc, but print fatal error.
another option is to mark FIFO element as "no-skb" which will make the sender stop until there is an skb